docs/rancher/csi-driver.md

```diff
@@ -40,65 +40,108 @@ When spinning up a Kubernetes cluster using Rancher RKE2 node driver, the Harves
 
 ![select-harvester-cloud-provider](/img/v1.2/rancher/rke2-cloud-provider.png)
 
-You can also deploy Harvester CSI Driver manually if you have already deployed the Harvester node driver manually. Perform the following steps to manually deploy the Harvester CSI Driver. 
+Suppose you would like to deploy the Harvester CSI driver without the cloud provider manually. You could create the Kubernetes cluster with RKE2 node driver with the default cloud provider (in v2.6, and the previous version is `None`).
 
-1. Generate the addon config.
+![](/img/v1.2/rancher/non-harvester-cloud-provider.png)
 
-:::note
+Then, perform the following steps to deploy the Harvester CSI Driver manually.
+#### Prerequisites
 
-The script uses `kubectl` and `jq` to operate the Harvester cluster. The script needs access to the `Harvester Cluster` kubeconfig to work. The `<serviceaccount name>` is usually your guest cluster name, and the `<namespace>` needs to match the namespace of the guest cluster.
+The script uses `kubectl` and `jq` to operate the Harvester cluster. The script needs access to the `Harvester Cluster` kubeconfig to work. The `<serviceaccount name>` is usually your guest cluster name, and the `<namespace>` needs to match the namespace of the guest cluster. Referring to the following figure. The cluster name is the `<serviceaccount name>`, and the namespace is the `<namespace>`.
 
-:::
+![](/img/v1.2/rancher/creating_guest_cluster.png)
 
 - Install `jq`:
-```
-# apt install jq
-```
-
-- Set up `kubectl` and the kubeconfig file (the kubeconfig should be allow you to access the `Harvester Cluster`):
-```
-# export KUBECONFIG=kubeconfig
-# export PATH="${PATH}:/var/lib/rancher/rke2/bin"
-```
-
-You can generate the kubeconfig file using the [generate_addon_csi.sh](https://raw.githubusercontent.com/harvester/harvester-csi-driver/master/deploy/generate_addon_csi.sh) script. It is available on the [harvester/harvester-csi-driver](https://github.com/harvester/harvester-csi-driver) repo.
-
-```
-# ./generate_addon_csi.sh <serviceaccount name> <namespace>
-```
-Get the `addon` yaml from the output of the script above.
 
-![](/img/v1.2/rancher/csi_addon_yaml.png)
+    ```
+    # apt install jq
+    ```
 
-1. Create the addon config file and add the `addon` yaml from the output above.
-
-```
-# mkdir -p /var/lib/rancher/rke2/etc/config-files
-# vim /var/lib/rancher/rke2/etc/config-files/csi_addon.yaml
-```
-
-1. Install Harvester CSI Driver.
-
-Install `Harvester CSI Driver` from the Rancher marketplace.
-![](/img/v1.2/rancher/install_csi_rancher_marketplace.png)
+- Set up `kubectl` and the kubeconfig file (the kubeconfig should be allow you to access the `Harvester Cluster`):
```

Comment:
```suggestion
- Set up `kubectl` and the kubeconfig file (kubeconfig should allow you to access the `Harvester Cluster`):
```

Reply:
not sure if it means `kubeconfig should allow you to access` or `kubeconfig should be allowed to access`.